src/components/sidebar/SafeListItem/index.tsx

```diff
@@ -80,7 +82,13 @@ const SafeListItem = ({
         )
       }
     >
-      <Link href={{ pathname: AppRoutes.home, query: { safe: `${shortName}:${address}` } }} passHref>
+      <Link
+        href={{
+          pathname: !safeAddress ? AppRoutes.home : router.pathname,
```

Comment:
Why do we need to check for the presence of `safeAddress`? The Safe address (`address`) is always defined.

Reply:
Thats how we make sure that we are not on a safe-unspecific route like `/welcome` since `safeAddress` is the one from the URL. Otherwise switching the Safe does nothing and the user is stuck on that page.

Reply:
I just realized something like `/welcome?safe=xyz` would fall through this and get the user stuck as well. Maybe an array of routes makes more sense then.

Reply:
Its only the `welcome` page where its necessary because its the only place where we render the `SafeList` without the `Sidebar`. I moved this logic up into the `SafeList` so we don't have to repeat it.

Reply:
We would have to repeat the `isWelcomePage` condition which might change for `SafeList` in the future. We could only extract this condition and reuse it in both components but this way we also don't need the `router` dependency in every `SafeListItem`.